internal/commands/vaultsecrets/secrets/read.go

```diff
@@ -0,0 +1,145 @@
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Read secret's metadata:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets secret read "test_secret"
+				`),
+			},
+			{
+				Preamble: `Read plaintext secret:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets secret read "test_secret" --plaintext
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "NAME",
+					Documentation: "The name of the secret to read.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:          "plaintext",
+					Description:   "Display the secret's value in plaintext.",
+					Value:         flagvalue.Simple(false, &opts.OpenSecret),
+					IsBooleanFlag: true,
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.AppName = appName
+			opts.SecretName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return readRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+type ReadOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+
+	AppName       string
+	SecretName    string
+	OpenSecret    bool
+	PreviewClient preview_secret_service.ClientService
+	Client        secret_service.ClientService
+}
+
+func readRun(opts *ReadOpts) error {
+	if opts.OpenSecret {
+		return runOpenAppSecret(opts)
+	}
+
+	req := secret_service.NewGetAppSecretParamsWithContext(opts.Ctx)
+	req.LocationOrganizationID = opts.Profile.OrganizationID
+	req.LocationProjectID = opts.Profile.ProjectID
+	req.AppName = opts.AppName
+	req.SecretName = opts.SecretName
+
+	resp, err := opts.Client.GetAppSecret(req, nil)
+	if err != nil {
+		return fmt.Errorf("failed to read the secret %q: %w", opts.SecretName, err)
+	}
+	return opts.Output.Display(newDisplayer(true).Secrets(resp.Payload.Secret))
+}
+
+func runOpenAppSecret(opts *ReadOpts) error {
+	// Table format doesn't handle large secret values well. So until that is fixed,
+	// we should not allow table format for this command.
+	if opts.Output.GetFormat() == format.Table {
+		return fmt.Errorf("table format is not supported for this command")
+	}
+
+	req := secret_service.NewOpenAppSecretParamsWithContext(opts.Ctx)
+	req.LocationOrganizationID = opts.Profile.OrganizationID
+	req.LocationProjectID = opts.Profile.ProjectID
+	req.AppName = opts.AppName
+	req.SecretName = opts.SecretName
+
+	resp, err := opts.Client.OpenAppSecret(req, nil)
+	if err != nil {
+		return fmt.Errorf("failed to read the secret %q: %w", opts.SecretName, err)
+	}
+
+	if opts.Output.GetFormat() == format.Unset {
+		fmt.Fprintf(opts.IO.Err(), resp.Payload.Secret.Version.Value)
+		return nil
+	}
+
+	d := newDisplayer(true).OpenAppSecrets(resp.Payload.Secret)
+	d.AddFields([]format.Field{
+		{
+			Name:        "Value",
+			ValueFormat: "{{.Version.Value}}",
+		},
+	})
```

Comment:
Hm this doesn't feel right to me. There is no way for a user to specify format.Unset. So there is no way to get to this behavior if the profile has a default output format set.

Reply:
hmmn.. can format be set on profile as well? in my usual runs, it is hitting that block - so without user specifying it, we hit that in the base case. what am i missing?

Reply:
Ok, there is  `"core/output_format": {"pretty", "table", "json"}`. I think we should discuss this 1-1.

Reply:
cc: @mercedesbh